samples/IISSample/Startup.cs

```diff
@@ -12,6 +13,7 @@ public class Startup
     {
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddAuthentication();
```

Comment:
You also need to add the auth middleware if you want HttpContext.User to get set without calling Authenticate, right? E.g. this sample doesn't work right now does it? Oh, the auth middelware is in the security repo, you can't reference it here can you?

HttpSysServer doesn't have this problem because it sets User before the app is invoked. What if the IISMiddleware always set the User? That code in GetUser should be run on every request regardless to clean up handles.